src/operations/indexes.ts

```diff
@@ -196,7 +219,25 @@ export class CreateIndexesOperation<
     this.options = options ?? {};
     this.collectionName = collectionName;
 
-    this.indexes = indexes;
+    // Ensure we generate the correct name if the parameter is not set
+    const namedIndexes = [];
+    for (const userIndex of indexes) {
+      const index: Omit<IndexDescription, 'key'> & { key: Map<string, IndexDirection> } = {
+        ...userIndex,
+        // Ensure the key is a Map to preserve index key ordering
+        key: userIndex.key instanceof Map ? userIndex.key : new Map(Object.entries(userIndex.key))
+      };
+      if (index.name == null) {
+        // Ensure every index is named
+        const keys = [];
+        for (const [name, direction] of index.key) {
+          keys.push(`${name}_${direction}`);
+        }
+        index.name = keys.join('_');
+      }
+      namedIndexes.push(index);
+    }
+    this.indexes = namedIndexes;
```

Comment:
Since we aren't changing the length or rearranging the indexes, I feel like this might be more straightforwardly expressed as
`this.indexes = indexes.map( userIndex => { ... return index; } )`

Reply:
done, it's still pulled out to a helper method but perhaps the code is streamlined enough to be in the constructor again. Thoughts @baileympearson?

Reply:
I'm fine either way